quantum/process_keycode/process_autocorrect.c

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 Google LLC
+// Copyright 2021 @filterpaper
+// SPDX-License-Identifier: Apache-2.0
+// Original source: https://getreuer.info/posts/keyboards/autocorrection
+
+#include "process_autocorrect.h"
+#include <string.h>
+#include "keycode_config.h"
+
+#ifndef AUTOCORRECT_DATA_H
+#    define AUTOCORRECT_DATA_H "autocorrect_data.h"
+#endif
+
+#if __has_include(AUTOCORRECT_DATA_H)
+#    include AUTOCORRECT_DATA_H
+#else
+#    pragma message "Warning! Autocorrect is using the default library!"
+#    include "autocorrect_data_default.h"
+#endif
+
+/**
+ * @brief handler for determining if autocorrect should process keypress
+ *
+ * @param keycode Keycode registered by matrix press, per keymap
+ * @param record keyrecord_t structure
+ * @param typo_buffer_size passed along to allow resetting of autocorrect buffer
+ * @param mods allow processing of mod status
+ * @return true Allow autocorection
+ * @return false Stop processing and escape from autocorrect.
+ */
+__attribute__((weak)) bool process_autocorrect_user(uint16_t *keycode, keyrecord_t *record, uint8_t *typo_buffer_size, uint8_t *mods) {
+    // See quantum_keycodes.h for reference on these matched ranges.
+    switch (*keycode) {
+        // Exclude these keycodes from processing.
+        case KC_LSFT:
+        case KC_RSFT:
+        case KC_CAPS:
+        case QK_TO ... QK_ONE_SHOT_LAYER_MAX:
+        case QK_LAYER_TAP_TOGGLE ... QK_LAYER_MOD_MAX:
+        case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
+            return false;
+
```

Comment:
```suggestion
            return false;
#ifndef NO_ACTION_ONESHOT
        case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
            return false;
#endif
```

Reply:
nah, should bo okay.  

If oneshot keys are disabled here, then it excludes them from being processed still.  Which I think is better behavior.  In fact....